x/launch/simulation/simulation.go

```diff
@@ -34,11 +34,23 @@ func SimulateMsgCreateChain(ak types.AccountKeeper, bk types.BankKeeper, k keepe
 			// No message if no coordinator
 			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgCreateChain, "skip create a new chain"), nil, nil
 		}
+
 		// skip if account cannot cover creation fee
 		creationFee := k.ChainCreationFee(ctx)
-		if !creationFee.Empty() && !bk.SpendableCoins(ctx, simAccount.Address).IsAllGTE(creationFee) {
-			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgCreateChain, "skip campaign creation"), nil, nil
+		customFee, err := simtypes.RandomFees(r, ctx, creationFee)
+		if err != nil {
+			return simtypes.NoOpMsg(
+					types.ModuleName,
+					types.TypeMsgCreateChain,
+					"skip chain creation"),
+				nil,
+				nil
```

Comment:
I'm wondering, shouldn't it be an actual error at this point? is RandomFees supposed to fail under some circumstances based on the state of the chain?

Reply:
that's a good point.  This is an error that has nothing to do with the simulation / state of the chain, so it should be passed out of the function.